src/sql/workbench/browser/modelComponents/tabbedPanel.component.ts

```diff
@@ -125,4 +127,19 @@ export default class TabbedPanelComponent extends ContainerBase<TabConfig> imple
 	onItemLayoutUpdated(item: ItemDescriptor<TabConfig>): void {
 		this._panel.updateTab(item.config.id, { title: item.config.title, iconClass: item.config.icon ? createIconCssClass(item.config.icon) : undefined });
 	}
+
+	public doAction(action: string, ...args: any[]): void {
+		switch (action) {
+			case ModelViewAction.SelectTab:
+				if (typeof args?.[0] !== 'string') {
```

Comment:
i am not sure about how panel component is handling the scenario where the id doesn't exist, now that this is exposed to extension authors, please make sure that is handled properly.

Reply:
It just ignores it. That behavior seems fine to me - we could look into throwing an error or something but I think the case of that happening is pretty low generally so don't really see the need to do anything more.

I'm going to look into adding tests but angular testing isn't something we have set up right now so I'm going to need to dig into that a bit and will work on that on the side.